File: CommonDataModel/src/main/java/com/kmitl/b_mw/common/nodeModel/Network.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.kmitl.b_mw.common.nodeModel;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 *
 * @author B_MW (Wai & May)
 */
public class Network extends BasicNode {

    final String nodeType = "Network";
    List<String> connect = new ArrayList<>();

    public List<String> getConnect() {
        return connect;
    }

    public void setConnect(List<String> connect) {
        this.connect = connect;
    }

}

File: Reader/lib/model/common/Network.js
var BasicNode = require("./BasicNode");

module.exports = (function () {
  function Network (id, key, name, validFrom, validTo) {
    BasicNode.call(this, id, key, name ,"Network", validFrom, validTo);
    this.connectWith = [];
  }

  Network.prototype = BasicNode.prototype;
  Network.prototype.getConnect = getConnect;
  Network.prototype.setConnect = setConnect;

  function getConnect () {
      return this.connectWith;
  }

  function setConnect (connect) {
      this.connectWith = connectWith;
  }

  return Network;
})();
